bigquery/google/cloud/bigquery/job.py

```diff
@@ -1418,6 +1418,21 @@ def num_dml_affected_rows(self):
         query_stats = self._query_statistics()
         return query_stats.get('numDmlAffectedRows')
 
+    @property
+    def undeclared_query_paramters(self):
+        """Return undeclared query parameters from job statistics, if present.
+
+        See:
+        https://cloud.google.com/bigquery/docs/reference/rest/v2/jobs#statistics.query.undeclaredQueryParamters
+
+        :rtype: list of dict
```

Comment:
I'd prefer to have this return a list of AbstractQueryParameter https://github.com/GoogleCloudPlatform/google-cloud-python/blob/master/bigquery/google/cloud/bigquery/_helpers.py#L362